src/adapters/store.js

```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
 import {version} from '../../config/constants.yml'
 import ExtendedString from "../libs/string";
 
-let moduleConfig = nconf.get().store
+export default class Store {
 
-const AbStore = require(`../libs/${moduleConfig.name}`)
-const abstractStore = new AbStore(moduleConfig.endpoint)
+  constructor() {
+    // get store from window if already initialized
+    if (window.__store) {
```

Comment:
ok for me, but i need to explore a little this solution ( could be better than mine ;) )

Reply:
after a little of reading, i'm completely ok with the approach, it's fully correct !
but 
the singleton can be recognized by it's signature (getInstance) i let you decide which one you prefer
by my side i have learnt something

Reply:
As you have pointed out this implementation hides the fact that Store is a singleton, but the user of the class is able to just call 'new Store()' to get the Store instance.
I guess the two approaches are valid, I will leave it as it is, we can change it later if needed.